design/Forest_Plot_design.Rmd

```diff
@@ -0,0 +1,421 @@
+```
+
+# Background 
+
+Design forest plot to visualize the direction and magnitude of the unadjusted and adjusted results for each category of endpoints.
+We would have a separate forest plot including all the time to event endpoints, binary endpoints etc.
+In the same forest plot we could consider adding the estimates of all the different approaches (e.g. sandwich estimator, bootstrap).
+
+In this example I have used the overall survival and more endpoints could be added (e.g. PFS etc.) in the same plot.
+An error bar with a diamond will be presented to indicate the point estimate with the corresponding 95% confidence interval for the OS unadjusted and adjusted results. Next to the forest plot and in the same line with the error bars we will present two columns that will show the coresponding HR [95% CI] and corresponding p-value. 
+
+
+# Steps
+
+- Load data 
+- Modify the report_table function in the matching_unanchored_tte.R file to get the results with a separate column for the HR and the 95% CI
+- In the maic_tte_unanchor function of the matching_unanchored_tte.R file, call the modified report_table so that the modified results are printed
+- Run the modified maic_tte_unanchor function to get the unadjusted and adjusted results 
+- Use the results to create a forest plot 
+
+
+## Load Data
+```{r}
+devtools::load_all()
+#### load data ----------------------------------------------------------
+
+### IPD
+# Read in relevant ADaM data and rename variables of interest
+adsl <- read.csv(system.file("extdata", "adsl.csv",
+  package = "maicplus",
+  mustWork = TRUE
+))
+adrs <- read.csv(system.file("extdata", "adrs.csv",
+  package = "maicplus",
+  mustWork = TRUE
+))
+adtte <- read.csv(system.file("extdata", "adtte.csv",
+  package = "maicplus",
+  mustWork = TRUE
+))
+
+### AgD
+# Baseline aggregate data for the comparator population
+target_pop <- read.csv(system.file("extdata", "aggregate_data_example_1.csv",
+  package = "maicplus", mustWork = TRUE
+))
+
+
+# for time-to-event endpoints, pseudo IPD from digitalized KM
+pseudo_ipd <- read.csv(system.file("extdata", "psuedo_IPD.csv",
+  package = "maicplus",
+  mustWork = TRUE
+))
+
+#### prepare data ----------------------------------------------------------
+target_pop <- process_agd(target_pop)
+adsl <- dummize_ipd(adsl, dummize_cols = c("SEX"), dummize_ref_level = c("Female"))
+use_adsl <- center_ipd(ipd = adsl, agd = target_pop)
+
+match_res <- estimate_weights(
+  data = use_adsl,
+  centered_colnames = grep("_CENTERED$", names(use_adsl)),
+  start_val = 0,
+  method = "BFGS"
+)
+```
+
+
+
+## Add report_table_fp in the matching_unanchored_tte.R file to separate HR from CI 
+```{r}
+##### ADDED
+report_table_fp <- function(coxobj, medSurvobj, tag = NULL) {
+  hr_res <- format(round(summary(coxobj)$conf.int[-2], 2), nsmall = 2)
+  # hr_res <- paste0(hr_res[1], "[", hr_res[2], ";", hr_res[3], "]")      #### DELETED
+  hr_pval <- format(round(summary(coxobj)$waldtest[3], 3), nsmall = 3)
```

Comment:
What would be the impact of changing this in the existing `report_table()`? KM plot? other tables?

I'm not sure if we need to create a whole extra function and table just to separate the estimate and the CI. We could paste them back together if necessary.

Reply:
I think this would not impact the KM plot; for tables we will just have HR and 95% CIs in two different columns but I don't think that this would be an issue. Indeed, maybe we don't need a new extra function to separate the estimate and CI. Maybe we could just modify the current teport_table() and paste them back if it is necessary.